atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/sweep/BackgroundSweeperImpl.java

```diff
@@ -175,23 +119,28 @@ public void run() {
 
     // Returns milliseconds to sleep
     @VisibleForTesting
-    long grabLocksAndRun(SweepLocks locks) throws InterruptedException {
+    long checkConfigAndRunSweep(SweepLocks locks) throws InterruptedException {
+        if (isSweepEnabled.get()) {
+            return grabLocksAndRun(locks);
+        } else {
+            log.debug("Skipping sweep because it is currently disabled.");
+        }
+        return 20 * (1000 + sweepPauseMillis.get());
```

Comment:
nit: probably best to extract this and the other `20*(1000 + sweepPauseMillis.get())` to something like `getBackoffTimeOnFailure()`.

The expression seems a bit arbitrary, though I guess there must have been reasons for its choice - and in any case it's the same as existing behaviour